src/window/mod.rs

```diff
@@ -701,6 +701,52 @@ pub(crate) fn reduce_window(
     }
 }
 
+/// Shim function for [`crate::dataflow::Dataflow::fold_window`].
+pub(crate) fn fold_window(
+    builder: &TdPyCallable,
+    folder: &TdPyCallable,
+    key: &StateKey,
+    acc: Option<TdPyAny>,
+    next_value: Option<TdPyAny>,
+) -> (Option<TdPyAny>, impl IntoIterator<Item = TdPyAny>) {
+    match next_value {
+        Some(value) => {
+            Python::with_gil(|py| {
+                let updated_acc: TdPyAny = match acc {
+                    // If there's no previous state for this key, use
+                    // the current value.
+                    None => {
+                        let initial_acc: TdPyAny =
+                            with_traceback!(py, builder.call1(py, (value.clone_ref(py),))).into();
```

Comment:
Re: my mistake. To match the semantics of the `stateful_map` operator (which also has a `builder`), the builder should be called with the `key` and not the `value`.

Reply:
Yep, fixing it.